Assets/AnkrSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs

```diff
@@ -6,42 +6,34 @@
 using AnkrSDK.WalletConnectSharp.Core;
 using AnkrSDK.WalletConnectSharp.Core.Models;
 using AnkrSDK.WalletConnectSharp.Core.Network;
+using AnkrSDK.WalletConnectSharp.Unity.Infrastructure;
 using AnkrSDK.WalletConnectSharp.Unity.Models.DeepLink;
 using AnkrSDK.WalletConnectSharp.Unity.Models.DeepLink.Helpers;
 using AnkrSDK.WalletConnectSharp.Unity.Network;
 using AnkrSDK.WalletConnectSharp.Unity.Utils;
-using Cysharp.Threading.Tasks;
 using UnityEngine;
+using Cysharp.Threading.Tasks;
 using Logger = AnkrSDK.InternalUtils.Logger;
 
 namespace AnkrSDK.WalletConnectSharp.Unity
 {
-	[RequireComponent(typeof(NativeWebSocketTransport))]
-	public class WalletConnect : MonoBehaviour
+	public class WalletConnect : IQuittableComponent, IPausableComponent, IDisposable
 	{
-		[SerializeField] private Wallets _defaultWallet = Wallets.MetaMask;
-		[SerializeField] private bool _autoSaveAndResume = true;
-		[SerializeField] private bool _connectOnAwake;
-		[SerializeField] private bool _connectOnStart = true;
-		[SerializeField] private bool _createNewSessionOnSessionDisconnect = true;
-		[SerializeField] private int _connectSessionRetryCount = 3;
-		[SerializeField] private string _customBridgeUrl = "https://testbridge.yartu.io/";
-		[SerializeField] private int _chainId = 1;
-		[SerializeField] private ClientMeta _appData = new ClientMeta();
-		[SerializeField] private NativeWebSocketTransport _transport;
-
-		[SerializeField] private WalletConnectEventWithSessionData _connectedEventSession;
-		[SerializeField] private WalletConnectEventWithSession _disconnectedEvent;
-		[SerializeField] private WalletConnectEventWithSession _newSessionConnected;
-		[SerializeField] private WalletConnectEventWithSession _resumedSessionConnected;
+		private readonly NativeWebSocketTransport _transport = new NativeWebSocketTransport();
+		private WalletConnectSettingsSO _settings;
+		private bool _initialized = false;
+
+		private readonly WalletConnectEventWithSessionData _connectedEventSession = new WalletConnectEventWithSessionData();
```

Comment:
As we are not using the editor anymore to configure events, we can get rid of these event classes and replace them with event Action<>. Would be much readable

Reply:
I will replace all of these events with a single one in the next phase of refactoring, first we need to have a general discussion of events and public properties model